specification/compute/resource-manager/Microsoft.Compute/ComputeRP/stable/2023-09-01/computeRPCommon.json

```diff
@@ -0,0 +1,2027 @@
+        },
+        "diskControllerType": {
+          "$ref": "#/definitions/DiskControllerType",
+          "description": "Specifies the disk controller type configured for the VM. **Note:** This property will be set to the default disk controller type if not specified provided virtual machine is being created with 'hyperVGeneration' set to V2 based on the capabilities of the operating system disk and VM size from the the specified minimum api version. You need to deallocate the VM before updating its disk controller type unless you are updating the VM size in the VM configuration which implicitly deallocates and reallocates the VM. Minimum api-version: 2022-08-01."
+        }
+      },
+      "description": "Specifies the storage settings for the virtual machine disks."
+    },
+    "UefiSettings": {
+      "properties": {
+        "secureBootEnabled": {
+          "type": "boolean",
+          "description": "Specifies whether secure boot should be enabled on the virtual machine. Minimum api-version: 2020-12-01."
+        },
+        "vTpmEnabled": {
+          "type": "boolean",
+          "description": "Specifies whether vTPM should be enabled on the virtual machine. Minimum api-version: 2020-12-01."
+        }
+      },
+      "description": "Specifies the security settings like secure boot and vTPM used while creating the virtual machine. Minimum api-version: 2020-12-01."
+    },
+    "EncryptionIdentity": {
+      "type": "object",
+      "properties": {
+        "userAssignedIdentityResourceId": {
+          "type": "string",
+          "description": "Specifies ARM Resource ID of one of the user identities associated with the VM."
+        }
+      },
+      "description": "Specifies the Managed Identity used by ADE to get access token for keyvault operations."
+    },
+    "ProxyAgentSettings": {
+      "type": "object",
+      "properties": {
+        "enabled": {
+          "type": "boolean",
+          "description": "Specifies whether ProxyAgent feature should be enabled on the virtual machine or virtual machine scale set."
+        },
+        "mode": {
+          "type": "string",
+          "description": "Specifies the mode that ProxyAgent will execute on if the feature is enabled. ProxyAgent will start to audit or monitor but not enforce access control over requests to host endpoints in Audit mode, while in Enforce mode it will enforce access control. The default value is Enforce mode.",
+          "enum": [
+            "Audit",
+            "Enforce"
+          ],
+          "x-ms-enum": {
+            "name": "Mode",
+            "modelAsString": true
+          }
+        },
+        "keyIncarnationId": {
+          "type": "integer",
+          "format": "int32",
```

Comment:
>int32 [](http://example.com/codeflow?start=21&length=5)

Would it be a good idea to make it int64 to reduce risk of wraparound?

Reply:
@hmyan90 Could you see this comment about keyIncarnationId?

Reply:
per @hmyan90, feature owner: "Changing the backend to be 64 would be too much - this keyIncarnationId is expected to be a rarely used property - it's designed for immediately problem mitigation."

we will move on with int32